### PR TITLE
ARROW-18392: [Python] Fix test_s3fs_wrong_region; set anonymous=True

### PR DESCRIPTION
**Pull request by milesgranger**
Will fix [ARROW-18392](https://issues.apache.org/jira/browse/ARROW-18392)

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit test-conda-python-*-pandas-*

File: python/pyarrow/tests/test_fs.py [hunk 1]
```diff
@@ -1321,22 +1321,23 @@ def test_s3fs_wrong_region():
     from pyarrow.fs import S3FileSystem
 
     # wrong region for bucket
-    fs = S3FileSystem(region='eu-north-1')
+    fs = S3FileSystem(region='eu-north-1', anonymous=True)
```
Comment (assignUser) [inline python/pyarrow/tests/test_fs.py:1324]: I'd add a comment explaining why this is important. Do we have other tests s3 tests that could also need this?

**Comment by raulcd** [comment]
@github-actions crossbow submit test-conda-python-*-pandas-*

File: python/pyarrow/tests/test_fs.py (hunk 1 above)
Comment (milesgranger) [reply python/pyarrow/tests/test_fs.py:1325]: Done, and I found only one other test that uses a real AWS call, aptly named `test_s3_real_aws` but that already sets `anonymous=True`. All others appear to use the endpoint override / minio.

**Comment by raulcd** [comment]
The pandas-0.24 job failure is expected to fail and should be fixed on: https://github.com/apache/arrow/pull/14714
The pandas nightly and upstream_devel failures are unrelated and I think should be fixed on its own ticket. I have opened this ticket to track them: https://issues.apache.org/jira/browse/ARROW-18394

File: python/pyarrow/tests/test_fs.py [hunk 2]
```diff
@@ -1321,22 +1321,24 @@ def test_s3fs_wrong_region():
...
-    with pytest.raises(OSError, match=msg) as exc:
+           "'eu-north-1' while the bucket is located in 'us-east-2'.")
+    with pytest.raises(OSError) as exc:
         fs.get_file_info("voltrondata-labs-datasets")
 
-    # Sometimes fails on unrelated network error, so next call would also fail.
-    if 'NETWORK_CONNECTION' in str(exc.value):
+    # Sometimes fails on unrelated network or permission error, so next
+    # call would also fail.
+    if 'UNKNOWN' not in str(exc.value):
```
Comment (pitrou) [inline python/pyarrow/tests/test_fs.py:1336]: Why "not in"? Don't we want this condition to be a bit more specific?

Comment (milesgranger) [reply python/pyarrow/tests/test_fs.py:1336]: Ya, good point. Since the main issue was `anonymous=True` needing to be set, I'll reset everything else.

**Review by pitrou — APPROVED** [review]
+1, thank you @milesgranger

**Pull request merged**